### PR TITLE
Add monthly update for July 2022

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

**Comment by pedroerp** [comment]
Maybe from now on we could start to publish these monthly updates to the discussion/announcements session as well?

> https://github.com/facebookincubator/velox/discussions/categories/announcements

File: velox/docs/monthly-updates/july-2022.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Documentation
+=============
+
+* Add documentation for Presto related :doc:`/functions/json`.
+
+Core Library
+============
+
+* Add `ApproxMostFrequentStreamSummary` that approximately computes the top frequent values from a large stream.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:13]: This is an implementation detail of the approx_most_frequent aggregate function and is not part of the public API. Perhaps, remove this.

File: velox/docs/monthly-updates/july-2022.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* Add support for cast from a JSON type value to ARRAYS or MAPS of JSON type values.
+* Add Parquet writer implemented over Apache Arrow Parquet writer.
+* Add NegatedBytesRange filter.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:16]: Perhaps, clarify this a bit.

`* Add NegatedBytesRange filter to optimize NOT IN ('a', 'b', 'c') filters on string columns pushed down into DWRF reader.`

File: velox/docs/monthly-updates/july-2022.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* Add PlanNode, PlanBuilder, registry support for Window operators.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:17]: typo: operators -> operator

Perhaps, clarify a bit.

`* Add WindowNode to describe window functions computations. Window operator implementation is still in development.`

File: velox/docs/monthly-updates/july-2022.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+* Add support for DECIMAL type Variant values.
+
+Presto Functions
+================
+
+* Add :func:`array_sort`, :func:`combinations` functions.
+* Add :func:`approx_most_frequent` aggregate function.
+* Extend aggregate functions :func:`min_by`, :func:`max_by` to support VARCHAR.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:25]: nit: perhaps, clarify: "... to support VARCHAR inputs."

File: velox/docs/monthly-updates/july-2022.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Hive Connector
+==============
+
+* Add NegatedBigintRange filters support to DWRF reader.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:30]: There is a duplicate item in Core Library section. Let's remove it from there and maybe clarify this one here:

```
* Add NegatedBytesRange filter to optimize NOT IN ('a', 'b', 'c') filters on string columns pushed down into DWRF reader.
```

File: velox/docs/monthly-updates/july-2022.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+=============
+
+* Add documentation for Presto related :doc:`/functions/json`.
+
+Core Library
+============
+
+* Add `ApproxMostFrequentStreamSummary` that approximately computes the top frequent values from a large stream.
+* Add support for cast from a JSON type value to ARRAYS or MAPS of JSON type values.
+* Add Parquet writer implemented over Apache Arrow Parquet writer.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:15]: Let's move this to Hive Connector section.

File: velox/docs/monthly-updates/july-2022.rst [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+Documentation
+=============
+
+* Add documentation for Presto related :doc:`/functions/json`.
+
+Core Library
+============
+
+* Add `ApproxMostFrequentStreamSummary` that approximately computes the top frequent values from a large stream.
+* Add support for cast from a JSON type value to ARRAYS or MAPS of JSON type values.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:14]: It feels like this belongs to Presto functions section.

File: velox/docs/monthly-updates/july-2022.rst [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+===========================
+
+* Add q5, q10, q12, q14, q19, q22 to TPC-H benchmark.
+* Optimize KLL sketch memory usage by allocating memory based on the number of elements retained.
+* Optimize arithmetic operations that check for overflow by enabling a fast execution path.
+* Optimize :func:`row_constructor` Presto function by removing redundant copy.
+* Add benchmarks for NegatedBigintRange, NegatedBytesRange filters.
+* Improve test coverage for various aggregate functions.
+* Improve performance of MmapAllocator::allocate.
+* Create nightly fuzzer run.
```
Comment (mbasmanova) [inline velox/docs/monthly-updates/july-2022.rst:42]: Can we add a link here?

**Review by mbasmanova — COMMENTED** [review]
@majetideepak Deepak, thank you for putting together this update. Looks great % a few comments.

File: velox/docs/monthly-updates/july-2022.rst (hunk 8 above)
Comment (majetideepak) [reply velox/docs/monthly-updates/july-2022.rst]: done.

File: velox/docs/monthly-updates/july-2022.rst (hunk 5 above)
Comment (majetideepak) [reply velox/docs/monthly-updates/july-2022.rst]: I just used `NOT IN` since we mention `string columns`.

**Review by mbasmanova — APPROVED** [review]
@majetideepak Thank you, Deepak.

**Pull request closed without merging**